### PR TITLE
Feature/history

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# ❓What kind of change does this PR introduce?

- [x] When the user clicks on a video, the video is will be appended to the history state.
- [x] User can delete an individual video from history.
- [x] User can delete all videos from history.

*Note you must be logged in to access the history feature*
Suggestions on the styling of the page are welcomed. As I feel the website is lacking better styles

### CODE REVIEW

File: src/reducer/historyReducer.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const initialHistoryState = { history: [] };
+const historyReducer = (state, { type, payload }) => {
+  console.log(payload);
```
Comment (AnkurChunekar) [inline src/reducer/historyReducer.js:3]: Remove console logs

**Comment by AnkurChunekar** [comment]
kevin pr
There's an error in the console, please look into it.

**Comment by AnkurChunekar** [comment]
You can remove the delete button if history has 0 videos, because in this case it fires an unnecessary network call.

**Comment by AnkurChunekar** [comment]
kevin pr
If I watch same video twice a new network request is fired, which tries to add a same video again to History which results in an error. Maybe you can have a simple check if video is already present or not.

File: src/service/addToHistory.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    });
+    console.log(response);
+    if (response.status === 201) {
+      historyDispatch({
+        type: 'ADD_TO_HISTORY',
+        payload: response.data.history,
+      });
+    }
+  } catch (err) {
+    console.log(err);
```
Comment (AnkurChunekar) [inline src/service/addToHistory.js]: You can use `console.error()` here.

File: src/service/addToHistory.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const addToHistory = async (video, token, historyDispatch) => {
+  try {
+    const response = await axios({
+      method: 'POST',
+      url: '/api/user/history',
+      data: { video },
+      headers: { authorization: token },
+    });
+    console.log(response);
```
Comment (AnkurChunekar) [inline src/service/addToHistory.js]: Remove this one too

File: src/service/deleteFromHistory.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      url: `/api/user/history/${_id}`,
+      headers: { authorization: token },
+    });
+    console.log(response);
+    historyDispatch({
+      type: 'DELETE_FROM_HISTORY',
+      payload: response.data.history,
+    });
+  } catch (err) {
+    console.log(err.response);
```
Comment (AnkurChunekar) [inline src/service/deleteFromHistory.js]: You can check the response status here, before calling the dispatch function.

**Review by AnkurChunekar — APPROVED** [review]
Rest everything looks good to me! good job @Kevin-Solomon

**Review by mohammed-sas — APPROVED** [review]

**Pull request merged**